### ISSUE TITLE
[Feature Request] `MultiDiscreteTensorSpec`in DQN

### ISSUE THREAD

**Issue by matteobettini** [pytorch/rl#1077]
Currently, DQN modules and losses want to know 
```
 action_space (str, optional): The action space to be considered.
            Must be one of
            ``"one-hot"``, ``"mult_one_hot"``, ``"binary"`` or ``"categorical"``.
```

We need to ass multi categorical to this (MultiDiscreteTensorSpec) and make it work in the modules and losses.

Plus, would it be possible to drop this argument and infer it from the action spec?
It seems redundant to me that I need to feed my DQN loss both the spec and a string which repeats a part of the spec

**Comment by vmoens** [comment]
This choice is aimed at making the loss usable without requiring any other class from the lib.
If we want to infer it from the policy which contains spec, we're asking the user to:
- use the policy from torchrl 
- in turn use the spec from torchrl
- in turn use the env from torchrl

That goes against the "adopt one component and not all" principle we any to achieve. 
But we could accept a spec and a string in these classes though

**Comment by matteobettini** [comment]
could this be optional and if not present we attempt to look at the spec?

**Comment by vmoens** [comment]
The question is also how do you fetch it when using a regular TensorDictSequential (not SafeSequential) which has no concept of action-spec.
I guess an informative error message like "you must either pass a spec, a string or a policy with a spec" could help...

**Comment by vmoens** [comment]
Let's keep in mind that the use case we want to support is also DQLoss(nn.Module)
As long as this can be used without needing to learn anything about TensorDict I'm ok with whichever option.
Some time in the future I will do a tutorial "DQN (or else) without TensorDict" I think it'd be fun!

**Comment by matteobettini** [comment]
Yea something like that.
When you do not have the needed parameter, you try to find it as hard as you can and if in the end you do not find it you tell the user that error

**Comment by matteobettini** [comment]
So users of torch rl can avoid to use those parameters and external users just need to give them
